components/spiffs/esp_spiffs.c

```diff
@@ -520,8 +541,25 @@ static int vfs_spiffs_open(void* ctx, const char * path, int flags, int mode)
         SPIFFS_clearerr(efs->fs);
         return -1;
     }
+#ifdef CONFIG_SPIFFS_USE_DIR
+    spiffs_stat s;
+    int ret = SPIFFS_fstat(efs->fs, fd, &s);
+    if (ret == SPIFFS_OK) {
+        vfs_spiffs_meta_t * meta = (vfs_spiffs_meta_t *)&s.meta;
+        if (meta->type == SPIFFS_TYPE_DIR) {
+            // It is directory, cannot be opened
+            errno = EISDIR;
+            ret = SPIFFS_close(efs->fs, fd);
+            if (ret < 0) {
+                errno = spiffs_res_to_errno(SPIFFS_errno(efs->fs));
+                SPIFFS_clearerr(efs->fs);
+            }
+            return -1;
+        }
+    }
+#endif
```

Comment:
Here a pretty long chunk of code is surrounded with an ifdef, could you please add a `// CONFIG_SPIFFS_USE_DIR` comment at the end of `#endif` line as a visual aid to the reader? Same in a few places below.